src/Symfony/Component/Serializer/Normalizer/CacheableSupportsInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+/**
+ * Marker interface for normalizers and denormalizers that return
+ * an integer in their supports*() methods.
+ *
+ * By implementing this interface, the return value of the
+ * supports*() methods will be cached by type and format, based on the returned integer.
+ *
+ * @author Jeroen Spee <https://github.com/Jeroeny>
+ */
+interface CacheableSupportsInterface
+{
+    // -1 : no, never supports the $format+$type, cache it
+    // 0  : no, no cache
+    // 1  : yes, no cache
+    // 2  : yes, always supports the $format+$type, cache it
+
+    public const SUPPORT_NEVER = -1;
+    public const SUPPORT_NOT = 0;
+    public const SUPPORT = 1;
+    public const SUPPORT_ALWAYS = 2;
+}
```

Comment:
Alternatively, make the interface extend `interface CacheableSupportsInterface extends NormalizerInterface` to narrow the return type.

```suggestion
/** @return int **/
public function supportsNormalization(mixed $data, string $format = null /*, array $context = [] */);

}
```